charts/tenantfilesystem/templates/composition.yaml

```diff
@@ -0,0 +1,808 @@
+          toFieldPath: "spec.deletePolicy"
+        - type: ToCompositeFieldPath
+          fromFieldPath: "metadata.name"
+          toFieldPath: "status.roleName"
+        - type: FromEnvironmentFieldPath
+          fromFieldPath: "eksAccountID"
+          toFieldPath: "spec.forProvider.assumeRolePolicyDocument"
+          transforms:
+            - type: convert
+              convert:
+                toType: string
+            - type: string
+              string:
+                fmt: |
+                  {
+                    "Version": "2012-10-17",
+                    "Statement": [
+                      {
+                        "Effect": "Allow",
+                        "Principal": {
+                          "AWS": "arn:aws:iam::%s:root"
+                        },
+                        "Action": "sts:AssumeRole",
+                        "Condition": {}
+                      }
+                    ]
+                  }
+
+    - name: policy
+      base:
+        apiVersion: iam.aws.crossplane.io/v1beta1
+        kind: Policy
+        metadata:
+          labels:
+            crossplane/xrd-kind: xtenantfilesystem.wadtfy.bmwgroup.net
+            crossplane/xrd-helm_chart_version: {{ .Chart.Version }}
+        spec:
+          deletionPolicy: Delete
+          forProvider:
+            tags:
+              - key: "wadtfy.bmwgroup.net/xrd-type"
+                value: "XTenantFileSystem"
+              - key: "wadtfy.bmwgroup.net/xrd-version"
+                value: "v1alpha1"
+              - key: "wadtfy.bmwgroup.net/component-name"
+                value: ""
+              - key: "wadtfy.bmwgroup.net/cluster-account-id"
+                value: ""
+              - key: "wadtfy.bmwgroup.net/product-name"
+                value: ""
+            document: ""
+            name:
+          providerConfigRef:
+            name: ""
+      patches:
+        - type: PatchSet
+          patchSetName: "providerConfigRef"
+        - type: PatchSet
+          patchSetName: "tagsV2"
+        - type: CombineFromComposite
+          combine:
+            variables:
+              - fromFieldPath: "spec.product"
+              - fromFieldPath: "metadata.labels[crossplane.io/claim-namespace]"
+              - fromFieldPath: "metadata.labels[crossplane.io/claim-name]"
+            strategy: string
+            string:
+              fmt: "tenant-filesystem-%s-%s-%s-remote"
+          toFieldPath: "metadata.name"
+        - fromFieldPath: "spec.deletePolicy"
+          toFieldPath: "spec.deletePolicy"
+        - fromFieldPath: "metadata.name"
+          toFieldPath: "spec.forProvider.name"
+          transforms:
+            - type: string
+              string:
+                fmt: "%s-EFSDescribeMountTargetIAMPolicy"
+          policy:
+            fromFieldPath: Required
+        - type: CombineFromComposite
+          combine:
+            variables:
+              - fromFieldPath: "spec.region"
```

Comment:
read from tenant EnvironmentConfig and not from spec of XRD

Reply:
changed